odoo/addons/base/models/ir_cron.py

```diff
@@ -447,13 +452,16 @@ def _trigger_list(self, at_list):
             ats = ', '.join(map(str, at_list))
             _logger.debug("will execute '%s' at %s", self.sudo().name, ats)
 
-        if min(at_list) <= now:
+        if min(at_list) <= now or os.getenv('ODOO_NOTIFY_CRON_CHANGES'):
             self._cr.postcommit.add(self._notifydb)
 
     def _notifydb(self):
-        """ Wake up the cron workers """
+        """ Wake up the cron workers
+        The ODOO_NOTIFY_CRON_CHANGES environment variable allows to force the notifydb on both
+        ir_cron modification and on trigger creation (regardless of call_at)
+        """
         with odoo.sql_db.db_connect('postgres').cursor() as cr:
-            cr.execute('NOTIFY cron_trigger')
+            cr.execute('NOTIFY cron_trigger, %s', (self.env.cr.dbname,))
```

Comment:
You sneaky little sneak, that's what you really were after weren't you!

FWIW `cr.execute` can take a list of parameters and it's usually more readable / more reliable, especially when there's a single parameter (avoids the need for a trailing comma).